src/utils/SpokeUtils.ts

```diff
@@ -102,3 +103,66 @@ export function isZeroAddress(address: string): boolean {
 export function getMessageHash(message: string): string {
   return isMessageEmpty(message) ? ZERO_BYTES : keccak256(message as Hex);
 }
+
+export async function findInvalidFills(spokePoolClients: {
+  [chainId: number]: SpokePoolClient;
+}): Promise<InvalidFill[]> {
+  const invalidFills: InvalidFill[] = [];
+
+  // Iterate through each spoke pool client
+  for (const spokePoolClient of Object.values(spokePoolClients)) {
+    // Get all fills for this client
+    const fills = spokePoolClient.getFills();
+
+    // Process each fill
+    for (const fill of fills) {
+      // Skip fills with unsafe deposit IDs
+      if (isUnsafeDepositId(fill.depositId)) {
+        continue;
+      }
+
+      // Get all deposits (including duplicates) for this fill's depositId, both in memory and on-chain
+      const depositResult = await spokePoolClients[fill.originChainId].findAllDeposits(fill.depositId);
```

Comment:
nb. RE mis-specified chain IDs, there might be something to consider here:

```suggestion
      const depositResult = await spokePoolClients[fill.originChainId]?.findAllDeposits(fill.depositId);
```